docs/runtimes/notifications.md

```diff
@@ -0,0 +1,131 @@
+However, there are several special cases where the notification is sent locally either way.
+These cases are:
+- Pipelines: To conserve backwards compatibility, the SDK sends the notifications as it did before adding the run
+  notifications mechanism. This means you need to watch the pipeline in order for its notifications to be sent.
+- Dask: Dask runs are always local (against a remote dask cluster), so the notifications are sent locally as well.
+
+## Notification Params and Secrets
+The notification parameters might contain sensitive information (slack webhook, git token, etc.). For this reason, 
+when a notification is created its params are masked in a kubernetes secret. The secret is named 
+`<run-uid>-<notification-id>` (or `<schedule-name>-<notification-id>`) and is created in the namespace where mlrun is 
+installed. In the notification params the secret reference is stored under the `secret` key once masked.
+
+## Notification Kinds
+
+Currently, the supported notification kinds and their params are as follows:
+
+- `slack`:
+  - `webhook`: The slack webhook to which to send the notification.
+- `git`:
+  - `token`: The git token to use for the git notification.
+  - `repo`: The git repo to which to send the notification.
+  - `issue`: The git issue to which to send the notification.
+  - `merge_request`: In gitlab (as opposed to github), merge requests and issues are separate entities. 
+                     If using merge request, the issue will be ignored, and vice versa.
+  - `server`: The git server to which to send the notification.
+  - `gitlab`: (bool) Whether the git server is gitlab or not.
+- `console` (no params, local only)
+- `ipython` (no params, local only)
+
+## Configuring Notifications For Runs
+
+In any `run` method you can configure the notifications via their model. For example:
+
+```python
+notification = mlrun.model.Notification(
+    kind="slack",
+    when=["completed","error"],
+    name="notification-1",
+    message="completed",
+    severity="info",
+    params={"webhook": "<slack webhook url>"}
+)
+function.run(handler=handler, notifications=[notification])
+```
+
+## Configuring Notifications For Pipelines
+For pipelines, you configure the notifications on the project notifiers. For example:
+
+```python
+project.notifiers.add_notification(notification_type="slack",params={"webhook":"<slack webhook url>"})
+```
+Instead of passing the webhook in the notification params, it is also possible in a Jupyter notebook to use the ` %env` 
+magic command:
+```
+%env SLACK_WEBHOOK=<slack webhook url>
+```
+
+## Setting Notifications on Live Runs
+You can set notifications on live runs via the `set_run_notifications` method. For example:
+
+```python
+import mlrun
+
+mlrun.get_run_db().set_run_notifications("<project-name>", "<run-uid>", [notification1, notification2])
+```
+
+Note that this overrides any existing notifications on the run. So deleting all notifications is done by passing an empty list.
```

Comment:
Note that this overrides any existing notifications on the run. To delete all notifications, pass an empty list.